src/pratt2.rs

```diff
@@ -0,0 +1,296 @@
+#![allow(missing_docs)]
+
+use super::*;
+
+trait Operator<'a, I, O, E>
+where
+    I: Input<'a>,
+    E: ParserExtra<'a, I>,
+{
+    type Op;
+    type OpParser: Parser<'a, I, Self::Op, E>;
+    const INFIX: bool = false;
+    const PREFIX: bool = false;
+
+    fn op_parser(&self) -> &Self::OpParser;
+    fn associativity(&self) -> Associativity;
+    fn fold_infix(&self, lhs: O, op: Self::Op, rhs: O) -> O {
+        unreachable!()
+    }
+    fn fold_prefix(&self, op: Self::Op, rhs: O) -> O {
+        unreachable!()
+    }
```

Comment:
Assuming this doesn't get implemented in a further commit: any particular reason for the `unimplemented!()` as opposed to just leaving the function impl up to the item implementing the trait?